src/index.js

```diff
@@ -9,21 +9,23 @@ function yieldTypes(schema) {
   return schema.types;
 }
 
-function filterTypes(whitelist) {
+function filterTypes(whitelistConfig) {
   return function(types) {
-    if (!whitelist) {
+    if (!whitelistConfig) {
       return types;
     }
 
+    const whitelistTypes = Object.getOwnPropertyNames(whitelistConfig);
```

Comment:
oh cool! I think I must have read something not entirely right at some point. Will update.